tests/test-customize-changesets-controller.php

```diff
@@ -1384,55 +1299,41 @@ public function test_update_item_edit_not_future_date_with_past_date() {
 		) );
 		$response = $this->server->dispatch( $request );
 
-		$this->assertErrorResponse( 'not_future_date', $response );
+		$this->assertErrorResponse( 'rest_invalid_param', $response );
 		$this->assertSame( $date_before, get_post( $manager->changeset_post_id() )->post_date_gmt );
 	}
 
 	/**
 	 * Test that update_item() rejects scheduling a changeset when it has a past date.
+	 *
+	 * @todo Verify that this is true -- not clear why should be be rejected.
 	 */
 	public function test_update_item_not_future_date_with_future_status() {
 		wp_set_current_user( self::$admin_id );
 
-		$manager = new WP_Customize_Manager();
-		$manager->save_changeset_post( array(
-			'date_gmt' => date( 'Y-m-d H:i:s', strtotime( '-1 year' ) ),
+		$uuid = wp_generate_uuid4();
+		$this->factory()->post->create( array(
+			'post_type' => 'customize_changeset',
+			'post_name' => $uuid,
+			'post_status' => 'auto-draft',
+			'post_content' => '{}',
 		) );
 
-		$status_before = get_post_status( $manager->changeset_post_id() );
-
-		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid ) );
 		$request->set_body_params( array(
 			'status' => 'future',
 		) );
 		$response = $this->server->dispatch( $request );
 
 		$this->assertErrorResponse( 'not_future_date', $response );
-		$this->assertSame( $status_before, get_post_status( $manager->changeset_post_id() ) );
-	}
-
-	/**
-	 * Test that update_item() rejects creating an auto-draft changeset with a date.
-	 */
-	public function test_update_item_create_cannot_supply_date_for_auto_draft_changeset() {
-		wp_set_current_user( self::$admin_id );
-
-		$uuid = wp_generate_uuid4();
-
-		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid ) );
-		$request->set_body_params( array(
-			'date_gmt' => strtotime( '+1 week' ),
-		) );
-		$response = $this->server->dispatch( $request );
-
-		$this->assertErrorResponse( 'cannot_supply_date_for_auto_draft_changeset', $response );
-
-		$manager = new WP_Customize_Manager();
-		$this->assertNull( $manager->find_changeset_post_id( $uuid ) );
+		$data = $response->get_data();
+		$this->assertSame( 'auto-draft', $data['post_status'] );
 	}
 
 	/**
 	 * Test that update_item() rejects editing an auto-draft changeset with a date.
+	 *
+	 * @todo Verify that it shouldn't be allowed.
```

Comment:
Good question. An `auto-draft` changeset in core is designed to be temporal. If it is never transitioned to another status, then it gets garbage-collected after 7 days. Every time a changeset is modified, its date gets bumped to the present datetime so that it will not lapse into 7 days old and thus get garbage-collected. For more info, see https://core.trac.wordpress.org/ticket/39713

Essentially, I think the idea is that `post_date` is a “managed” readonly field until the status transitions from `auto-draft` to something else, because at that point the post is no longer in danger of being automatically garbage-collected.